modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/GetObjectRelationshipEdgeCandidatesMVCResourceCommandTest.java

```diff
@@ -0,0 +1,358 @@
+						).put(
+							"objectRelationshipId",
+							_objectRelationshipAA_AAA.getObjectRelationshipId()
+						)
+					).put(
+						JSONUtil.put(
+							"label",
+							_getEdgeLabel(
+								_objectDefinitionA, _objectRelationshipA_AA)
+						).put(
+							"objectRelationshipId",
+							_objectRelationshipA_AA.getObjectRelationshipId()
+						)
+					)
+				).put(
+					"label",
+					_getEdgeLabel(
+						_objectDefinitionAAA, objectRelationshipAAA_AAAA)
+				).put(
+					"objectRelationshipId",
+					objectRelationshipAAA_AAAA.getObjectRelationshipId()
+				).put(
+					"root", false
+				)
+			).toString(),
+			_getJSONArray(
+				0, objectDefinitionAAAA.getObjectDefinitionId()
+			).toString());
+
+		// Get object relationship edges candidates where there is a parent who
+		// is inside a hierarchical structure but the parent depth plus child
+		// depth is greater than 4
+
+		Assert.assertEquals(
+			_jsonFactory.createJSONArray(
+			).toString(),
+			_getJSONArray(
+				3, objectDefinitionAAAA.getObjectDefinitionId()
+			).toString());
+
+		_objectRelationshipLocalService.deleteObjectRelationship(
+			objectRelationshipAAA_AAAA.getObjectRelationshipId());
+
+		// Get object relationship edges candidates where there is a parent who
+		// is not inside a hierarchical structure
+
+		ObjectDefinition objectDefinitionBBB = _createObjectDefinition(
+			false, "BBBB");
+
+		ObjectRelationship objectRelationshipBBB_AAAA = _relateObjectDefinition(
+			false, objectDefinitionBBB, objectDefinitionAAAA);
+
+		Assert.assertEquals(
+			_jsonFactory.createJSONArray(
+			).put(
+				JSONUtil.put(
+					"label",
+					_getEdgeLabel(
+						objectDefinitionBBB, objectRelationshipBBB_AAAA)
+				).put(
+					"objectRelationshipId",
+					objectRelationshipBBB_AAAA.getObjectRelationshipId()
+				)
+			).toString(),
+			_getJSONArray(
+				0, objectDefinitionAAAA.getObjectDefinitionId()
+			).toString());
+
+		_objectRelationshipLocalService.deleteObjectRelationship(
+			objectRelationshipBBB_AAAA.getObjectRelationshipId());
+
+		// Get object relationship edges candidates where there is a parent who
+		// is inside a hierarchical structure and is the root
+
+		ObjectRelationship objectRelationshipA_AAAA = _relateObjectDefinition(
+			false, _objectDefinitionA, objectDefinitionAAAA);
+
+		Assert.assertEquals(
+			_jsonFactory.createJSONArray(
+			).put(
+				JSONUtil.put(
+					"ancestors", _jsonFactory.createJSONArray()
+				).put(
+					"label",
+					_getEdgeLabel(_objectDefinitionA, objectRelationshipA_AAAA)
+				).put(
+					"objectRelationshipId",
+					objectRelationshipA_AAAA.getObjectRelationshipId()
+				).put(
+					"root", true
+				)
+			).toString(),
+			_getJSONArray(
+				0, objectDefinitionAAAA.getObjectDefinitionId()
+			).toString());
+	}
+
+	private static ObjectDefinition _createObjectDefinition(
+			boolean nested, String objectDefinitionName)
+		throws Exception {
+
+		ObjectDefinition objectDefinition =
```

Comment:
@MarinhoFeliphe isolate this method

Reply:
Done [here](https://github.com/liferay-objects/liferay-portal/pull/2431/commits/265dfbe8829261a9b02817d901b07a9a10a641de)